### PR TITLE
TRUNK-5838: Improving the validateJavaVersion() method

### PR DESCRIPTION
**Pull request by sacull**
Now the if condition checks for all java versions
not supported by Spring Framework 5. Also, update docs.

## Description of what I changed

The Spring Framework version of OpenMRS requires java version 1.8 or higher. The java version check method should therefore only allow these versions. I have corrected the regular expression checking the java version to prevent it from using versions 1.6 and 1.7. In addition, I added a test checking the correctness of operation for version 10+.

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5838

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/util/OpenmrsUtil.java [hunk 1]
```diff
@@ -1160,14 +1160,14 @@ public static String getOpenmrsLogLocation() {
...
-	 * @throws ApplicationContextException if the current JVM version is earlier than Java 6
+	 * @throws ApplicationContextException if the current JVM version is earlier than Java 8
 	 */
 	public static void validateJavaVersion() {
-		// check whether the current JVM version is at least Java 6
-		if (JdkVersion.getJavaVersion().matches("1.(0|1|2|3|4|5).(.*)")) {
-			throw new APIException("OpenMRS requires Java 6, but is running under " + JdkVersion.getJavaVersion());
+		// check whether the current JVM version is at least Java 8
+		if (JdkVersion.getJavaVersion().matches("1.(0|1|2|3|4|5|6|7).(.*)")) {
+			throw new APIException("OpenMRS requires Java 8, but is running under " + JdkVersion.getJavaVersion());
```
Comment (HerbertYiga) [inline api/src/main/java/org/openmrs/util/OpenmrsUtil.java:1170]: Does openmrs require only java 8,as per the exception message here?

Comment (sacull) [reply api/src/main/java/org/openmrs/util/OpenmrsUtil.java:1170]: "OpenMRS requires Java 8 or later, but is running under " - My English is not the best. Will this version of the message be better?

Comment (HerbertYiga) [reply api/src/main/java/org/openmrs/util/OpenmrsUtil.java:1170]: we could use (OpenMRS requires Java 8 and above,but it is running under)

Comment (sacull) [reply api/src/main/java/org/openmrs/util/OpenmrsUtil.java]: Done. Thank you for the hint.

**Comment by dkayiwa** [comment]
@sacull did you get a chance to look at this? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

File: api/src/main/java/org/openmrs/util/OpenmrsUtil.java (hunk 1 above)
Comment (sherrif10) [reply api/src/main/java/org/openmrs/util/OpenmrsUtil.java]: @sacull Can you squash your commits into one as suggested by @dkayiwa  in thee link he provided

Comment (dkayiwa) [reply api/src/main/java/org/openmrs/util/OpenmrsUtil.java]: @sherrif10 did you see this? https://talk.openmrs.org/t/rfc-favour-not-squashing-commits-over-squashing-commits/28482

**Comment by sacull** [comment]
@dkayiwa @sherrif10 I made 'merge' based on https://wiki.openmrs.org/display/docs/Git-Squashing+commits. There were three, now there are five. I still need to learn how to do 'merge'. :/

I try to make one of these five as soon as I learn how to do it.

File: api/src/main/java/org/openmrs/util/OpenmrsUtil.java (hunk 1 above)
Comment (sherrif10) [reply api/src/main/java/org/openmrs/util/OpenmrsUtil.java]: @dkayiwa  sure i went through it,However i may come back to you with any suggestion

**Comment by sacull** [comment]
Pull request closed due to commit merge problem. Current pull request -> https://github.com/openmrs/openmrs-core/pull/3324

**Pull request closed without merging**